core/state/state_object.go

```diff
@@ -269,16 +255,15 @@ func (s *stateObject) SetState(db Database, key, value common.Hash) {
 // lookup only happens in the fake state storage.
 //
 // Note this function should only be used for debugging purpose.
-func (s *stateObject) SetStorage(storage map[common.Hash]common.Hash) {
-	// Allocate fake storage if it's nil.
-	if s.fakeStorage == nil {
-		s.fakeStorage = make(Storage)
-	}
-	for key, value := range storage {
-		s.fakeStorage[key] = value
+//
+func (s *stateObject) SetStorage(db Database, storage map[common.Hash]common.Hash) {
+	// We set the root to empty root, to make prevent GetCommittedState from
+	// reading the original data, it should thus only have access to what is
+	// set in the loop below.
+	s.data.Root = emptyRoot
```

Comment:
Hmm, it's not enough. State can still be hit in snapshot. e.g. https://github.com/ethereum/go-ethereum/blob/master/core/state/state_object.go#L216

Reply:
No, I think we're executing snapshot-free when we're tracing